spec/syntax.md

```diff
@@ -88,127 +88,212 @@ The syntax specification takes into account the following design restrictions:
->declension (using either a fixed dictionary, algorithmic declension, ML, etc.):
->
->```
->{Hello, {$userName :person case=vocative}!}
->```
+> **Example**
+> In a Java `.properties` file, the values `hello` and `hello2` both contain
+> an identical _message_ which consists of a single _pattern_.
+> This _pattern_ consists of _text_ with exactly three spaces before and after the word "Hello":
+> ```properties
+> hello = {   Hello   }
+> hello2={   Hello   }
+> ```
 
->A _message_ with a `$userObj` _variable_ formatted with
->the custom `:person` _function_ capable of
->plucking the first name from the object representing a person:
->
->```
->{Hello, {$userObj :person firstName=long}!}
->```
+```abnf
+text = 1*(text-char / text-escape)
+text-char = %x0-5B         ; omit \
+          / %x5D-7A        ; omit {
+          / %x7C           ; omit }
+          / %x7E-D7FF      ; omit surrogates
+          / %xE000-10FFFF
+```
 
-_Functions_ can be _standalone_, or can be an _opening element_ or _closing element_.
+### Placeholder
 
-A **_standalone_** _function_ is not expected to be paired with another _function_.
-An **_opening element_** is a _function_ that SHOULD be paired with a _closing function_.
-A **_closing element_** is a _function_ that SHOULD be paired with an _opening function_.
+A **_placeholder_** is an _expression_ that appears inside of a _pattern_
+and which will be replaced during the formatting of a _message_.
 
-An _opening element_ MAY be present in a message without a corresponding _closing element_,
-and vice versa.
+```abnf
+placeholder = expression
+```
 
->A message with two markup-like _functions_, `button` and `link`,
->which the runtime can use to construct a document tree structure for a UI framework:
->
->```
->{{+button}Submit{-button} or {+link}cancel{-link}.}
->```
+## Matcher
 
+A **_matcher_** is the _body_ of a _message_ that allows runtime selection
+of the _pattern_ to use for formatting.
+This allows the form or content of a _message_ to vary based on values
+determined at runtime.
 
-### Selection
+A _matcher_ consists of the keyword `match` followed by at least one _selector_
+and at least one _variant_.
 
-A **_selector_** selects a specific _pattern_ from a list of available _patterns_
-in a _message_ based on the value of its _expression_.
-A message can have multiple selectors.
+When the _matcher_ is processed, the result will be a single _pattern_ that serves 
+as the template for the formatting process.
 
->A message with a single _selector_, `{$count :number}`. `:number` is a built-in function. 
->
->```
->match {$count :number}
->when 1 {You have one notification.}
->when * {You have {$count} notifications.}
->```
+A _message_ can only be considered _valid_ if the following requirements are
+satisfied:
+* The number of _keys_ on each _variant_ MUST be equal to the number of _selectors_.
+* At least one _variant_ MUST exist whose _keys_ are all equal to the "catch-all" key `*`.
```

Comment:
Running Prettier on this will replace the `*` with `-` and add a space before the list. Might be easier to just use its styling for markdown?

Reply:
Good callout and one for the style guide.